cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -228,16 +238,64 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
+     *  sanityCheckGoalNames() method (L118)  don't fail if a user submits custom default goals that have less members then the default
+     *  anomaly.detection.goals.
+     * @param configuration The configuration instance to be checked.
+     * @throws UnsupportedOperationException If the configuration contains self.healing.goals configurations.
+     */
+    public static void checkGoals(CruiseControlConfiguration configuration) {
+        // If self healing goals are defined then these take precedence
+        if (configuration.getConfigOption(CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY) != null) {
+            String selfHealingGoalsString = configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY);
+            List<String> selfHealingGoals = Arrays.asList(selfHealingGoalsString.split("\\s*,\\s*"));
+            if (!selfHealingGoals.isEmpty()) {
+                throw new UnsupportedOperationException("Cruise Control's self healing functionality is not currently supported. Please remove " +
+                        CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY + " config");
+            }
+        }
+
+        // Either self.healing.goals is null or it is an empty list
+        String anomalyGoalsString;
+        if (configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY) != null) {
+            // The user has defined some anomaly detection goals
+            anomalyGoalsString = configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY);
+        } else {
+            // No anomaly detection goals have been defined by the user, therefore the defaults defined in Cruise Control
+            // will be used.
+            anomalyGoalsString = CRUISE_CONTROL_DEFAULT_ANOMALY_DETECTION_GOALS;
+        }
+
+        Set<String> anomalyDetectionGoals = new HashSet<>(Arrays.asList(anomalyGoalsString.split("\\s*,\\s*")));
+
+        String defaultGoalsString = configuration.getConfigOption(CRUISE_CONTROL_DEFAULT_GOALS_CONFIG_KEY);
+        Set<String> defaultGoals = new HashSet<>(Arrays.asList(defaultGoalsString.split("\\s*,\\s*")));
+
+        // Remove all the goals which are present in the default goals set from the anomaly detection goals
+        anomalyDetectionGoals.removeAll(defaultGoals);
+
+        if (anomalyDetectionGoals.size() > 0) {
+            // If the anomaly detection goals contain goals which are not in the default goals then the CC startup
+            // checks will fail, so we make the anomaly goals match the default goals
+            configuration.setConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY, defaultGoalsString);
+            log.warn("Anomaly goals contained goals which are not in the configured default goals. Anomaly goals have " +
+                    "been changed to match the specified default goals.");
+        }
+    }
```

Comment:
So this area of code is to prevent users from misconfiguring `goals` and `anomaly.detection.goals` by catching the error, notifying the user, and then setting a sane configuration. This should be taken care of on the CC side but since it isn't I am glad it's taken care of here, it minimizes the configurations which the users have to think about when getting CC up and running.

Reply:
I think we might want to set this value back to the default `anomaly.detection.goals` instead of the default `goals` should the user misconfigure `anomaly.detection.goals`

Reply:
This area of code is to make sure a user defined `default.goals` config does not make teh CC config checks fail. 

We are preventing the user from configuring `anomaly.detection.goals` (at least we will be when I submit a PR for that --- good thing you made me check) so they can't be mis-configured. This is all about `default.goals` config and making sure the user can set it to what they want without CC failing to start.